etsd/core/views.py

```diff
@@ -1,35 +1,50 @@
-from etsd.users.utils import get_authority_users
+from django.http.response import HttpResponseRedirect
 from django.views.generic import UpdateView
 from authorities.models import Authority
 from .forms import AuthorityUsersModelForm
-from dal import autocomplete
-from etsd.users.models import User
-
-class UserAutocomplete(autocomplete.Select2QuerySetView):
-    def get_queryset(self):
-        # Don't forget to filter out results depending on the visitor !
-        if not self.request.user.is_authenticated:
-            return User.objects.none()
-
-        qs = User.objects.all()
-
-        if self.q:
-            qs = qs.filter(name__istartswith=self.q)
-
-        return qs
+from django.contrib import messages
+from django.utils.translation import ugettext as _
+from django.contrib.auth import get_user_model
+from django.contrib.auth.models import Permission
 
 
 class AuthorityEditUsersView(UpdateView, ):
     model = Authority
     form_class = AuthorityUsersModelForm
-    template_name='core/edit_users.html'
-    context_object_name = 'authority'
-
-    def get_initial(self):
-        return {"usrs": get_authority_users(self.object)}
-
+    template_name='core/edit_authority_data.html'
 
     def get_form_kwargs(self):
         kwargs = super(AuthorityEditUsersView, self).get_form_kwargs()
         kwargs['request'] = self.request
-        return kwargs
\ No newline at end of file
+        return kwargs
+
+    def form_valid(self,form):
+        usrs = form.cleaned_data['users']
+        auth = self.request.user.get_authority()
+        initial_users = get_user_model().objects.filter(authorities=auth)
+        added_users = set(usrs).difference(set(initial_users))
+        removed_users = set(initial_users).difference(set(usrs))
+        for usr in added_users:
+            if usr.authorities.exists():
```

Comment:
Since you do a validation here (if the user belongs to a different authority) it's better to moe that to the form's clean method.